federation/cmd/federation-apiserver/app/core.go

```diff
@@ -39,14 +38,15 @@ import (
 	namespaceetcd "k8s.io/kubernetes/pkg/registry/core/namespace/etcd"
 	secretetcd "k8s.io/kubernetes/pkg/registry/core/secret/etcd"
 	serviceetcd "k8s.io/kubernetes/pkg/registry/core/service/etcd"
+	"k8s.io/kubernetes/pkg/registry/generic"
 )
 
-func installCoreAPIs(s *options.ServerRunOptions, g *genericapiserver.GenericAPIServer, restOptionsFactory restOptionsFactory) {
-	serviceStore, serviceStatusStore := serviceetcd.NewREST(restOptionsFactory.NewFor(api.Resource("service")))
```

Comment:
oh... this is the federation-apiserver... it's going to be really painful to get the serviceetcd to use different etcd prefixes for different servers. opened https://github.com/kubernetes/kubernetes/issues/38262 to see if we can fix it. if not, federation will need to override the etcd location for services specially